app/src/docs/v1.api-spec.yaml

```diff
@@ -0,0 +1,2744 @@
+  /document/:
+    put:
+      summary: Create a document
+      description: >-
+        Creates a link between an activity and a previously existing object in
+        COMS.
+      operationId: createDocument
+      tags:
+        - Document
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Request-Document'
+      responses:
+        '201':
+          description: Document created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Document'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        "422":
+          $ref: "#/components/responses/UnprocessableEntity"
+        default:
+          $ref: '#/components/responses/Error'
+  /document/{documentId}:
+    delete:
+      summary: Delete a document by ID
+      description: Deletes the document associated with the provided document ID.
+      operationId: deleteDocument
+      tags:
+        - Document
+      parameters:
+        - $ref: '#/components/parameters/Path-DocumentId'
+      responses:
+        '200':
+          description: Document deleted successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Document'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        "422":
+          $ref: "#/components/responses/UnprocessableEntity"
+        default:
+          $ref: '#/components/responses/Error'
+  /document/list/{activityId}:
+    get:
+      summary: List documents by Activity ID
+      description: Retrieves a list of documents associated with a given activity.
+      operationId: listDocuments
+      tags:
+        - Document
+      parameters:
+        - $ref: '#/components/parameters/Path-ActivityId'
+      responses:
+        '200':
+          description: A list of documents for the specified activity
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/DB-Document'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        "422":
+          $ref: "#/components/responses/UnprocessableEntity"
+        default:
+          $ref: '#/components/responses/Error'
+  /enquiry/:
+    get:
+      summary: Gets a list of
+      description: >-
+        Gets a list of enquiries, if the query self is passed and true the list
+        returned will contain just the enquiries made by the user, as per RBAC.
```

Comment:
`self` is no longer a parameter after the implementation of the RBAC. Returned results will be based on scope attributes attached to the users groups.